File: v3/java/src/main/java/com/konfigthis/newscatcherapi/client/api/SearchLinkApi.java
package com.konfigthis.newscatcherapi.client.api;

import com.konfigthis.newscatcherapi.client.ApiClient;
import com.konfigthis.newscatcherapi.client.ApiException;
import com.konfigthis.newscatcherapi.client.model.DtoResponsesSearchResponseSearchResponse;
import java.util.List;

public class SearchLinkApi extends SearchLinkApiGenerated {

    public SearchLinkApi(ApiClient apiClient) throws IllegalArgumentException {
        super(apiClient);
    }
    public class GetRequestBuilder extends GetRequestBuilderGenerated {
        public GetRequestBuilder() {
            super();
        }

        public List<Object> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                DtoResponsesSearchResponseSearchResponse response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        DtoResponsesSearchResponseSearchResponse res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
    public class PostRequestBuilder extends PostRequestBuilderGenerated {
        public PostRequestBuilder() {
            super();
        }

        public List<Object> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                DtoResponsesSearchResponseSearchResponse response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        DtoResponsesSearchResponseSearchResponse res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
}

File: v3/csharp/src/Newscatcherapi.Net/Api/SearchLinkApi.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Newscatcherapi.Net.Client;
using Newscatcherapi.Net.Model;
using Newscatcherapi.Net.Client;

namespace Newscatcherapi.Net.Api
{
    public partial class SearchLinkApi : SearchLinkApiGenerated {
        public SearchLinkApi() : base() {}
        public SearchLinkApi(string basePath): base(basePath) {}
        public SearchLinkApi(Newscatcherapi.Net.Client.Configuration configuration): base(configuration) {}
        public SearchLinkApi(Newscatcherapi.Net.Client.ISynchronousClient client, Newscatcherapi.Net.Client.IAsynchronousClient asyncClient, Newscatcherapi.Net.Client.IReadableConfiguration configuration): base(client, asyncClient, configuration) {}
    
        public List<Object> GetAll(Dictionary<string, object> parameters = null) {
            parameters = parameters ?? new Dictionary<string, object>();
            int pageSize = parameters.ContainsKey("pageSize") ? (int) parameters["pageSize"] : 100;

            Type type = this.GetType();
            MethodInfo method = type.GetMethod("Get");
            List<object> paramValues = SearchAllHelper.prepareParameters(parameters, method);

            try {
                DtoResponsesSearchResponseSearchResponse response = (DtoResponsesSearchResponseSearchResponse) method.Invoke(this, paramValues.ToArray());
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        parameters["page"] = page;
                        DtoResponsesSearchResponseSearchResponse res = (DtoResponsesSearchResponseSearchResponse) method.Invoke(this, paramValues.ToArray());
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }

        public List<Object> PostAll(SearchURLRequest request) {
            int pageSize = request.PageSize;
            try {
                DtoResponsesSearchResponseSearchResponse response = this.Post(request);
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        request.Page = page;
                        DtoResponsesSearchResponseSearchResponse res = this.Post(request);
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }
    }
}